### ISSUE TITLE
Group Case Sensitivity Issue

### ISSUE THREAD

**Issue by tobias-g** [egnyte/gitlabform#160]
When a group name has caps in its URL but lowercase is used for `gitlabform` the group is found but no projects within it are processed.

## Steps to reproduce

1. create a group with caps e.g. `gitlabformtest/FOO`
> Note: you may also need to go to the group settings and change it in the group settings (General Settings > Advanced > Change group URL)
2. create a project in the group e.g. `foo-project`
3. setup gitlabform config group_settings for the group e.g. `gitlabformtest/foo`
4. run gitlabform output:
```
$ gitlabform gitlabformtest/foo
*** # of groups to process: 1
*** # of projects to process: 0
> (1/1) Processing: gitlabformtest/foo
GitLabForm version: 1.18.3.2 (the latest)
```

## Cause

It looks like the API call to find groups works fine (case insensitive) so the group is returned and the subsequent call to get the projects returns the projects for the group.

However then when the projects are added to the list to be processed the `startswith` check will not match anything as it uses the lowercase group name and `path_with_namespace` from the API call returns the namespace with uppercase values.

See https://github.com/egnyte/gitlabform/blob/f765571ce826b3d745471ff7fbbb293589f3083f/gitlabform/gitlab/groups.py#L52

## Workaround

Pretty simple workaround is to run `gitlabform gitlabformtest/FOO` instead of `gitlabform gitlabformtest/foo` but the fact it found the group but no projects stumped me for a while.

**Comment by lazouz** [comment]
Hi, I have a very similar bug : 

# Steps to reproduce

in my config.yml file I have a group called  : "CAPS", it's the name displayed in gitlab but the url is "/blablabla/caps"

When I call `gitlabform config CAPS`, no project is processed. 
But if I call  `gitlabform config caps` the projects inside the group and  are found and processed.

And if I put "caps" in the yaml config file, I get an error at the first project found in "caps" group : 
```
Traceback (most recent call last):
  File "/usr/local/bin/gitlabform", line 6, in <module>
    GitLabForm().main()
  File "/usr/local/lib/python3.9/site-packages/gitlabform/gitlabform/core.py", line 205, in main
    self.process_all(projects_and_groups, groups)
  File "/usr/local/lib/python3.9/site-packages/gitlabform/gitlabform/core.py", line 324, in process_all
    configuration = self.c.get_effective_config_for_project(project_and_group)
  File "/usr/local/lib/python3.9/site-packages/gitlabform/configuration/projects_and_groups.py", line 40, in get_effective_config_for_project
    common_and_group_config = self.merge_configs(common_config, group_config)
  File "/usr/local/lib/python3.9/site-packages/gitlabform/configuration/projects_and_groups.py", line 112, in merge_configs
    merged_config[key] = {
TypeError: 'bool' object is not a mapping
```

Also, if I use `gitlabform config ALL_DEFINED` no project inside my group "CAPS" is processed 

# Cause 
Do you need more info about this bug ? Does it seems really the same as the previously described bug ? 

# Workaround
use `gitlabform config caps` for all the groups containing capital letters, and not ALL_DEFINED (not very good workaround, but it works)

**Comment by gdubicki** [comment]
Hi @tobias-g ! Sorry for the late reply.

Hi @lazouz !

Both of your issues will be fixed in v2.0.0 which will be release within days from now.

In the mean time you can try out a pre-release that contains the fix, [v2.0.0rc2](https://github.com/egnyte/gitlabform/releases/tag/v2.0.0rc2).

**Comment by gdubicki** [comment]
v2 final has been released. Please reopen this ticket if you still have issues.
